host/distro/devtree.go

```diff
@@ -35,11 +40,32 @@ func DTCompatible() []string {
 	return dtCompatible
 }
 
+// DTRevision returns the device revision (e.g. a02082 for the Raspberry Pi 3)
+// from the Linux device tree, or 0 if the file is missing or malformed.
+func DTRevision() uint32 {
+	mu.Lock()
+	defer mu.Unlock()
+
+	if dtRevision != 0 {
+		return dtRevision
+	}
+	b, err := ioutil.ReadFile("/proc/device-tree/system/linux,revision")
+	if err != nil {
+		return 0
```

Comment:
The drawback is that the failure mode will continuously read the file and fail. For example the file doesn't exist on my desktop, so it means DTRevision() will try to open the file every time the function is called.

The two other functions in this file don't have this problem, so I'd prefer to have this fixed.

Reply:
Done.